vm/ubpf_vm.c

```diff
@@ -573,6 +573,24 @@ ubpf_exec(const struct ubpf_vm *vm, void *mem, size_t mem_len, uint64_t* bpf_ret
     }
 }
 
+int
+ubpf_run(struct ubpf_vm *vm, void *mem, size_t mem_len, bool jit,
+    uint64_t *bpf_return_value, char **errmsg)
+{
+    if (jit) {
+        ubpf_jit_fn fn = ubpf_compile(vm, errmsg);
+        if (fn == NULL) {
+            return -1;
+        }
+        *bpf_return_value = fn(mem, mem_len);
+    } else {
+        if (ubpf_exec(vm, mem, mem_len, bpf_return_value) < 0) {
+            *bpf_return_value = UINT64_MAX;
```

Comment:
@iomartin I set this as a generic message for now. We can revisit this at another time.

Reply:
Oh but this breaks nosetests. Some oddness here on how error printing is done.

Reply:
Oh right. Upon further examination, getting the tests to pass with those changes is not that trivial. The strings part are not hard (though we'd have to update a bunch of test cases), but the testing framework doesn't like the return -1 too.

I think we can make it work (I can help with the nosetests if needed), but if there is a simpler solution I'm not seeing, then I think that'd be preferred.

Reply:
@iomartin @lsgunth this is actually quite problematic since there are also a bunch of printf statements to stderr inside ubpf_exec. I am not sure how best to address this but it is proving problematic.

Reply:
I'd just leave it. Outside of scope for this PR. Seems like it could use a cleanup eventually, though.

Reply:
I will leave this out for now and file an issue after this PR gets in.